scripts/update_core_lowerbound.py

```diff
@@ -21,67 +20,69 @@ def scan_requirement(line, supported_versions):
-                            updated = True
+                        elif min_version > max(supported_versions):
+                            print("This requirement is newer than any (yet) supported branches.")
                         else:
-                            print(supported_versions)
                             print(
-                                "No supported lower bounds can satisfy requirement"
-                                " range, this branch can no longer be supported."
+                                f"No supported lower bounds {supported_versions} can satisfy"
+                                " requirement range {requirement.specifier}, this branch can no"
+                                " longer be supported."
                             )
                             exit(1)
                     break
-    return updated, new_requirement
+    return new_requirement
 
 
-def scan_pyproject_toml(supported_versions):
+def scan_pyproject_toml(supported_versions: set[Version]) -> None:
     changed = False
     with open("pyproject.toml") as fp:
         pyproject = tomlkit.load(fp)
-        requirements = pyproject["project"]["dependencies"]
+        requirements: list[str] = pyproject["project"]["dependencies"]
         for i, line in enumerate(requirements):
-            updated, new_requirement = scan_requirement(line, supported_versions)
-            if updated:
+            new_requirement = scan_requirement(line, supported_versions)
+            if new_requirement is not None:
                 requirements[i] = new_requirement
                 changed = True
     if changed:
         with open("pyproject.toml", "w") as fp:
             tomlkit.dump(pyproject, fp)
 
 
-def scan_requirements_txt(supported_versions):
+def scan_requirements_txt(supported_versions: set[Version]) -> None:
     changed = False
     with open("requirements.txt") as fp:
         lines = fp.readlines()
         for i, line in enumerate(lines):
             requirement, sep, comment = line.partition(" #")
-            updated, new_requirement = scan_requirement(requirement, supported_versions)
-            if updated:
+            new_requirement = scan_requirement(requirement, supported_versions)
+            if new_requirement is not None:
                 lines[i] = new_requirement + sep + comment
                 changed = True
     if changed:
         with open("requirements.txt", "w") as fp:
             fp.writelines(lines)
 
 
-def main():
-    request = requests.get(core_template_url)
+def main() -> None:
+    request = requests.get(CORE_TEMPLATE_URL)
     if request.status_code != 200:
         print("Failed to find supported branches, not checking lower bounds")
         exit(0)
 
     core_template = yaml.safe_load(request.content)
-    supported_versions = {parse(v) for v in core_template["supported_release_branches"]}
-    latest = core_template["latest_release_branch"]
```

Comment:
and this is the key that will keep us from updating to latest-released over and over until a new supported is added. got it.